### PR TITLE
test(integration): add integration test for not found psp in code review pipeline

### PR DESCRIPTION
**Pull request by infantesimone**
#### List of Changes

- add integration test for not found psp in code review pipeline

#### Motivation and Context

The goal of this PR is to add integration tests to check error messages in case of unavailable PSP like for "Marca da bollo" or for too high amounts. Tests were done for the `CARD` method that shows the error after entering the card data; test for `satispay` to test the message in the method selection page.

#### How Has This Been Tested?

integration test in code review pipeline

#### Screenshots (if appropriate):

Screenshot 2024-10-29 alle 22 59 54

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/__integration_tests__/paymentflow.integration.test.ts [hunk 1]
```diff
@@ -301,6 +314,72 @@ describe("Checkout fails to calculate fee", () => {
...
+    const pspNotFoundCtaId = "#pspNotFoundCtaId";
+    const pspNotFoundCtaElem = await page.waitForSelector(pspNotFoundCtaId);
+    const pspNotFoundCtaText = await pspNotFoundCtaElem.evaluate((el) => el.textContent);
+    expect(pspNotFoundCtaText).toContain(translation.pspUnavailable.cta.primary);
+
+    const errorDescriptionId = "#pspNotFoundBodyId";
+    const errorDescriptionElem = await page.waitForSelector(errorDescriptionId);
+    const errorDescriptionText = await errorDescriptionElem.evaluate((el) => el.textContent);
+    expect(errorDescriptionText).toContain(translation.pspUnavailable.body);
+
```
Comment (ciuffagianluca) [inline src/__integration_tests__/paymentflow.integration.test.ts:347]: Maybe here you can add a step to check if the cta button really lead user to payment choice page

File: src/__integration_tests__/paymentflow.integration.test.ts [hunk 2]
```diff
@@ -301,6 +314,72 @@ describe("Checkout fails to calculate fee", () => {
...
```
Comment (ciuffagianluca) [inline src/__integration_tests__/paymentflow.integration.test.ts:380]: Same as above, even if here the modal is shown on the payment choice page, so you should check if the page is the same.

File: src/__integration_tests__/paymentflow.integration.test.ts (hunk 1 above)
Comment (infantesimone) [reply src/__integration_tests__/paymentflow.integration.test.ts:347]: thanks! done

File: src/__integration_tests__/paymentflow.integration.test.ts (hunk 2 above)
Comment (infantesimone) [reply src/__integration_tests__/paymentflow.integration.test.ts]: done

**Review by ciuffagianluca — APPROVED** [review]

**Pull request merged**